src/examples/spaceshooter/renderer/WallModels.tsx

```diff
@@ -0,0 +1,65 @@
+import { useGLTF } from '@react-three/drei';
+import { memo, useEffect, useRef } from 'react';
+import * as THREE from 'three';
+import wall_cornerMediumGLTF from '../assets/wall_cornerMedium2.glb?url';
+import wall_largeGLTF from '../assets/wall_large2.glb?url';
+import wall_mediumGLTF from '../assets/wall_medium2.glb?url';
+import levelData from '../levels/level_1';
+
+export default memo(function WallModels() {
+    const { models } = levelData;
+
+    const { scene: wallMedium } = useGLTF(wall_mediumGLTF);
+    const { scene: wallLarge } = useGLTF(wall_largeGLTF);
+    const { scene: wallCornerMedium } = useGLTF(wall_cornerMediumGLTF);
+
+    const groupRef = useRef<THREE.Group>(null);
+
+    useEffect(() => {
+        if (groupRef.current) {
+            placeModels();
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    const placeModels = () => {

```

Comment:
instead of manually cloning and position these things, I think you should create R3F Components for each of the wall parts.... take a look at how the ship model is used

Reply:
Good point, addressed in latest commit: https://github.com/playmint/playerchain-demo/pull/83/commits/54f49b4458fd61d2bf2c7a335960641fde3e4381